storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCenter.java

```diff
@@ -0,0 +1,222 @@
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Hierarchy getHierarchyWithSubSystem(SubSystemType subSystem) {
+        return getHierarchyWithSubSystems(Arrays.asList(subSystem));
+    }
+
+    @Override
+    public Hierarchy getHierarchyWithSubSystems(List<SubSystemType> subSystems) {
+        List<Hierarchy> hierarchies = this.getHierarchies();
+        for (Hierarchy hierarchy : hierarchies) {
+            Hierarchy ret = hierarchy;
+            for (SubSystemType subSystem : subSystems) {
+                if (!hierarchy.getSubSystems().contains(subSystem)) {
+                    ret = null;
+                    break;
+                }
+            }
+            if (ret != null) {
+                return ret;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean isMounted(Hierarchy hierarchy) {
+
+        List<Hierarchy> hierarchies = this.getHierarchies();
+        if (Utils.CheckDirExists(hierarchy.getDir())) {
+            for (Hierarchy h : hierarchies) {
+                if (h.equals(hierarchy)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void mount(Hierarchy hierarchy) throws IOException {
+
+        if (this.isMounted(hierarchy)) {
+            LOG.error("{} is already mounted", hierarchy.getDir());
+            return;
+        }
+        Set<SubSystemType> subSystems = hierarchy.getSubSystems();
+        for (SubSystemType type : subSystems) {
+            if (this.getHierarchyWithSubSystem(type) != null) {
+                LOG.error("subSystem: {} is busy", type.name());
+                subSystems.remove(type);
+            }
+        }
+        if (subSystems.size() == 0) {
+            return;
+        }
+        if (!Utils.CheckDirExists(hierarchy.getDir())) {
+            new File(hierarchy.getDir()).mkdirs();
+        }
+        String subSystemsName = CgroupUtils.subSystemsToString(subSystems);
+        SystemOperation.mount(subSystemsName, hierarchy.getDir(), "cgroup", subSystemsName);
+
+    }
+
+    @Override
+    public void umount(Hierarchy hierarchy) throws IOException {
+        if (this.isMounted(hierarchy)) {
+            hierarchy.getRootCgroups().delete();
+            SystemOperation.umount(hierarchy.getDir());
+            CgroupUtils.deleteDir(hierarchy.getDir());
+        } else {
+            LOG.error("{} is not mounted", hierarchy.getDir());
+        }
+    }
+
+    @Override
+    public void createCgroup(CgroupCommon cgroup) throws SecurityException {
+        if (cgroup.isRoot()) {
+            LOG.error("You can't create rootCgroup in this function");
+            return;
+        }
+        CgroupCommon parent = cgroup.getParent();
+        while (parent != null) {
+            if (!Utils.CheckDirExists(parent.getDir())) {
+                LOG.error("Parent {} does not exist", parent.getDir());
+                return;
```

Comment:
I think printing out an error message is ok for now.  Since we are not sure if we want to crash the supervisor if something in cgroups goes wrong.  Though this is something to investigate in a follow jira